components/raftstore/src/store/peer.rs

```diff
@@ -1658,6 +1668,9 @@ where
                     // Init the in-memory pessimistic lock table when the peer becomes leader.
                     self.activate_in_memory_pessimistic_locks();
 
+                    self.prepare_merge_fence = 0;
+                    self.pending_prepare_merge = None;
```

Comment:
Perhaps we should clear them when the leader step down instead.

Reply:
Moved to `clear_in_memory_pessimistic_locks`